File: ghc-4335c07/libraries/text/benchmarks/ruby/sort.rb
#!/usr/bin/env ruby

require './utils.rb'

def sort(filename)
  File.open(filename, 'r:utf-8') do |file|
    content = file.read
    puts content.lines.sort.join
  end
end

ARGV.each do |f|
  t = benchmark { sort(f) }
  STDERR.puts "#{f}: #{t}"
end

File: ghc-4335c07/libraries/text/benchmarks/python/sort.py
#!/usr/bin/env python

import utils, sys, codecs

def sort(filename):
    content = open(filename, encoding='utf-8').read()
    lines = content.splitlines()
    lines.sort()
    print('\n'.join(lines))

for f in sys.argv[1:]:
    t = utils.benchmark(lambda: sort(f))
    sys.stderr.write('{0}: {1}\n'.format(f, t))
